### PR TITLE
Add Support for customData Object

### PR DESCRIPTION
**Pull request by dcpesses**
Adds the ability to pass a customData object, which can contain licensing server urls and authentication tokens, to the Receiver app when casting media content.

### CODE REVIEW

**Comment by jthomerson** [comment]
@onebytegone, I'll do a review of this, but can you test it out?

**Comment by jthomerson** [comment]
Actually, @onebytegone can you do the first review? I'm behind.

File: README.md [hunk 1]
```diff
@@ -172,6 +179,11 @@ subtitles = {
    'https://example.com/videos/video-2.mp4': 'Subtitle2',
 };
 
+customdata = {
+   'https://example.com/videos/video-1.mp4': {'customColor': '#0099ee'},
```
[the remark this answers is not in the document]
Comment (jthomerson) [reply README.md]: @onebytegone why didn't the CI build catch this? `grunt standards` should've failed there, right?

[the remark this answers is not in the document]
Comment (onebytegone) [reply README.md]: This is in the README example. At this time we are only linting JS files. Is it possible to lint code blocks in markdown? If so, that'd be quite nice to have.

[the remark this answers is not in the document]
Comment (jthomerson) [reply README.md]: LOL. Lost my context. Didn't see it was in README 🤦‍♂️

File: src/js/tech/ChromecastTech.js [hunk 2]
```diff
@@ -54,6 +54,7 @@ ChromecastTech = {
       this._hasPlayedAnyItem = false;
       this._requestTitle = options.requestTitleFn || _.noop;
       this._requestSubtitle = options.requestSubtitleFn || _.noop;
+      this._requestCustomData = options.requestCustomDataFn || {};
```
[the remark this answers is not in the document]
Comment (dcpesses) [reply src/js/tech/ChromecastTech.js]: Good catch! Pushing my changes now.

File: README.md [hunk 3]
```diff
@@ -172,6 +179,11 @@ subtitles = {
    'https://example.com/videos/video-2.mp4': 'Subtitle2',
 };
 
+customdata = {
```
[the remark this answers is not in the document]
Comment (dcpesses) [reply README.md]: Ah, must have skimmed too quickly through the coding standards the first time around. Thanks for pointing that out.

File: src/js/tech/ChromecastTech.js [hunk 4]
```diff
@@ -156,6 +157,7 @@ ChromecastTech = {
           mediaInfo = new chrome.cast.media.MediaInfo(source.src, source.type),
           title = this._requestTitle(source),
           subtitle = this._requestSubtitle(source),
+          customData = this._requestCustomData(source) || {},
```
Comment (onebytegone) [inline src/js/tech/ChromecastTech.js]: Does this need to have an empty object as a default? Doing so prevents `requestCustomDataFn` from intentionally sending a falsey value to `customData`. For example, a user of this plugin might want to send an empty string via `customData` (I'm not quite sure why they would need to do this, but I'd seen crazier things).

Comment (dcpesses) [reply src/js/tech/ChromecastTech.js]: Fair enough. I'll even take it a step further by only setting `mediaInfo.customData` if the user has passed `requestCustomDataFn`

**Comment by dcpesses** [comment]
@onebytegone Please take a look, I’ve made the suggested updates and rebased my commits as well.

**Comment by dcpesses** [comment]
@onebytegone @jthomerson Did I rebase & squash my commits properly? Do I need to make any additional changes to the code?

File: src/js/tech/ChromecastTech.js [hunk 5]
```diff
@@ -165,6 +167,9 @@ ChromecastTech = {
       mediaInfo.metadata.metadataType = chrome.cast.media.MetadataType.GENERIC;
       mediaInfo.metadata.title = title;
       mediaInfo.metadata.subtitle = subtitle;
+      if (customData) {
```
Comment (onebytegone) [inline src/js/tech/ChromecastTech.js]: It appears this still won't [allow for a falsey value to be passed via `customData`](https://github.com/silvermine/videojs-chromecast/pull/20#discussion_r184977990). Was there a particular reason you decided to go this route?

Comment (dcpesses) [reply src/js/tech/ChromecastTech.js]: IIRC, I believe it's possible that the `mediaInfo` object could already have an existing `customData` object, and I didn't want to overwrite any existing values if it wasn't set in the plugin's config on the page. (I remember seeing at one point that, even though there didn't appear to be a `customData` object set in `mediaInfo` when logged to the console locally, an object had indeed been passed to the receiver and logged to the remote console.)

Comment (dcpesses) [reply src/js/tech/ChromecastTech.js]: @onebytegone does this still need to be refactored, or can my previous comment justify merging this PR?

**Pull request merged**